stateth/stateth.go

```diff
@@ -0,0 +1,257 @@
+
+	se.gclient, err = gapi.New(fmt.Sprintf("%s:%s", grafanaUser, grafanaPass), fmt.Sprintf("http://localhost:%d", se.config.GrafanaPort))
+	if err != nil {
+		log.Warn(err.Error())
+		return nil
+	}
+	if err = se.importGrafanaDatasource(); err != nil {
+		return err
+	}
+	if err = se.importGrafanaDashboards(); err != nil {
+		return err
+	}
+	fmt.Println(fmt.Sprintf("grafana listening on http://localhost:%d", se.config.GrafanaPort))
+	fmt.Println(fmt.Sprintf("username: %s", grafanaUser))
+	fmt.Println(fmt.Sprintf("password: %s", grafanaPass))
+	fmt.Println()
+
+	return nil
+}
+
+func (se *Stateth) StopExternal() error { return se.cleanupContainers() }
+
+func (se *Stateth) runNetwork() error {
+	log.Info("creating docker network", "network", se.config.DockerPrefix)
+	command := strings.Split(fmt.Sprintf("docker network create %s", se.config.DockerPrefix), " ")
+	r, err := exec.Command(command[0], command[1:]...).CombinedOutput()
+	if err != nil {
+		log.Error(string(r))
+		return err
+	}
+	return nil
+}
+
+func (se *Stateth) runInfluxDB() error {
+	log.Info("pulling influxdb:1.5.2 docker image")
+	command := strings.Split("docker pull influxdb:1.5.2", " ")
+	r, err := exec.Command(command[0], command[1:]...).CombinedOutput()
+	if err != nil {
+		log.Error(err.Error())
+		return err
+	}
+
+	log.Info("running influxdb docker container", "container", fmt.Sprintf("%s_influxdb", se.config.DockerPrefix))
+	command = strings.Split(fmt.Sprintf("docker run --network %s --name %s_influxdb -e INFLUXDB_DB=metrics -e INFLUXDB_ADMIN_USER=%s -e INFLUXDB_ADMIN_PASSWORD=%s -p %d:8086 -d influxdb:1.5.2", se.config.DockerPrefix, se.config.DockerPrefix, influxDBAdminUser, influxDBAdminPass, se.config.InfluxDBPort), " ")
+	r, err = exec.Command(command[0], command[1:]...).CombinedOutput()
+	if err != nil {
+		log.Error(string(r))
+		return err
+	}
+	return nil
+}
+
+func (se *Stateth) runGrafana() error {
+	log.Info("pulling grafana/grafana:5.1.3 docker image")
+	command := strings.Split("docker pull grafana/grafana:5.1.3", " ")
+	r, err := exec.Command(command[0], command[1:]...).CombinedOutput()
+	if err != nil {
+		log.Error(string(r))
+		return err
+	}
+
+	log.Info("running grafana docker container", "container", fmt.Sprintf("%s_grafana", se.config.DockerPrefix))
+	//command = strings.Split(fmt.Sprintf("docker run --network %s --name=%s_grafana -p %d:3000 -d grafana/grafana:5.1.3", se.config.DockerPrefix, se.config.DockerPrefix, se.config.GrafanaPort), " ")
+	command = strings.Split(fmt.Sprintf("docker run --network %s --name=%s_grafana -e GF_AUTH_ANONYMOUS_ENABLED=true -p %d:3000 -d grafana/grafana:5.1.3", se.config.DockerPrefix, se.config.DockerPrefix, se.config.GrafanaPort), " ")
+	r, err = exec.Command(command[0], command[1:]...).CombinedOutput()
+	if err != nil {
+		log.Error(string(r))
+		return err
+	}
+	return nil
+}
+
+func (se *Stateth) cleanupContainers() error {
+	log.Info("removing influxdb container")
```

Comment:
The convention on Geth is to have capitalized logs. I kept all of the comments lowercase in `stateth`, but since we might be running this as part of Geth, we should adhere its conversions. Please change this (as well as other comments) to:
```
Removing InfluxDB container
```